### PR TITLE
marker rotation using magnetic heading

### PR DESCRIPTION
**Pull request by ChristianTovar**
# Demo 

https://github.com/opensailing/nautic_net_web/assets/13787741/060065ae-12a1-4466-a6a6-3642bb7ba1e5

### CODE REVIEW

File: assets/js/leaflet.js [hunk 1]
```diff
@@ -90,7 +93,7 @@ class BoatView {
     this.boatId = boatId;
     this.map = map;
     this.trackCoordinates = trackCoordinates;
-    this.marker = Leaflet.marker([0, 0]).addTo(map);
+    this.marker = Leaflet.marker([0, 0], { icon: boatIcon }).addTo(map);
```
Comment (polvalente) [inline assets/js/leaflet.js:96]:
```suggestion
    this.marker = Leaflet.marker([0, 0], { icon: boatIcon, color: trackColor }).addTo(map);
```

Does this work for coloring?

Comment (polvalente) [reply assets/js/leaflet.js:96]: Ah nevermind, I was looking at the customIconMarker, and it uses the color prop in a custom way :(
I wonder if you could get that passed down to the SVG somehow

Comment (polvalente) [reply assets/js/leaflet.js:96]: one way would be to use a data url and to interpolate it directly to the SVG string (encoding it as base64 afterwards), but that's kinda ugly.

File: priv/static/images/boat.svg
Comment (polvalente) [inline priv/static/images/boat.svg:1]: Do we still want to keep this?

Comment (ChristianTovar) [reply priv/static/images/boat.svg:1]: We might, for reference purposes I'd say

**Review by polvalente — APPROVED** [review]

**Pull request merged**